Dart/src/com/jetbrains/lang/dart/ide/formatter/DartSpacingProcessor.java

```diff
@@ -384,6 +388,25 @@ else if (type1 == REFERENCE_EXPRESSION) {
       return Spacing.createSpacing(0, 1, preserveNewline, true, 0);
     }
 
+    if (elementType == NAMED_ARGUMENT) {
+      if (type1 == COLON) {
+        return addSingleSpaceIf(true);
+      }
+      if (type2 == COLON) {
+        return noSpace();
+      }
+    }
+    if (elementType == TYPE_ARGUMENTS && type1 == LT || type2 == GT) {
```

Comment:
That's a typo, which should be: (type1 == LT || type2 == GT)
All type args should be formatted with minimal spaces, as in: 
abstract class HashMap<K, V> implements Map<K, V>
I'll fix it. It currently permits a space before '<' which should be tightened up, too.